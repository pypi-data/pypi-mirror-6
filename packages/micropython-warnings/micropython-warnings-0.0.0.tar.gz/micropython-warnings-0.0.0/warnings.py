# MicroPython doesn't support warnings
