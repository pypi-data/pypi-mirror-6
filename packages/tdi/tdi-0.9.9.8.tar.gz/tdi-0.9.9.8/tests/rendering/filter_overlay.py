#!/usr/bin/env python
import warnings as _warnings
_warnings.resetwarnings()
_warnings.filterwarnings('error')

from tdi import filters as _filters
from tdi import html as _html

class Count(_filters.BaseEventFilter):
    def __init__(self, builder):
        super(Count, self).__init__(builder)
        self.tags = {}

    def handle_starttag(self, name, attr, closed, data):
        self.tags[name] = self.tags.get(name, 0) + 1
        self.builder.handle_starttag(name, attr, closed, data)

    def finalize(self):
        keys = self.tags.keys()
        keys.sort()
        for key in keys:
            print "%s: %d" % (key, self.tags[key])
        return self.builder.finalize()

html = _html.replace(overlay_eventfilters=[Count])

print "Before Loading"
template = html.from_string("""
<anode tdi:overlay="foo"></anode>
""").overlay(html.from_string("""
<bnode tdi:overlay="foo">
<cnode />
</bnode>
"""))

print ">>> Between Loading and Rendering"

template.render()

print ">>> Between Rendering"

template.render()

print ">>> After Rendering"
