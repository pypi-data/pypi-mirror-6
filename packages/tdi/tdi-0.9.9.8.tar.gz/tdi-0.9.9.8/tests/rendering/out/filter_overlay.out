Before Loading
>>> Between Loading and Rendering
bnode: 1
cnode: 1

<bnode>
<cnode />
</bnode>
>>> Between Rendering

<bnode>
<cnode />
</bnode>
>>> After Rendering
